src/bar.js

```diff
@@ -186,15 +190,64 @@ export default class Bar {
             }
 
             this.gantt.unselect_all();
-            this.group.classList.toggle('active');
+            
+            //  add dependency by popup button
+            if(this.gantt.dependencyBar != null){
+            	this.add_dependency();
+            }else{
+                this.group.classList.toggle('active');
+                this.show_popup();
+            }
 
-            this.show_popup();
         });
     }
 
+    add_dependency(){
+		// already marked a dependency
+		var markedTask = this.gantt.dependencyBar.task;
+		if(markedTask == null)
+			return;
+		
+		var changedTask;
+		
+		// check if tasks are already connected
+		if(!this.task.dependencies.includes(markedTask.id) && !markedTask.dependencies.includes(this.task.id) && this.task !== markedTask){
+			// same start date no dependency
+			if(this.task._start.getTime() === markedTask._start.getTime()){
+	    		this.release_marked_bar();
+				return;
+			}
+			
+			// check which task starts later
+			if(this.task._start.getTime() > markedTask._start.getTime()){
+				changedTask = this.task;
+				this.task.dependencies.push(markedTask.id);
+			}else{
+				changedTask = markedTask;
+				markedTask.dependencies.push(this.task.id);
+			}
+		
+			// fire dependencyAdded event
+			this.gantt.trigger_event('dependency_added', [changedTask]);
+	      	// recalculate dependency tree
+			this.gantt.setup_dependencies();
+			// redraw gantt
+			this.gantt.render();
+		}
+		this.release_marked_bar();
+    }
+
+    release_marked_bar(){
+    	// remove class
+    	this.gantt.dependencyBar.group.classList.toggle('addArrow');
```

Comment:
Check if `dependencyBar` exists.

Also CSS classnames should be separated with a dash `-`, so this should be `add-arrow`, but `add-arrow` itself is not a good classname for what it does. Must be `selected-for-dependency` or something similar.